pkg/apis/componentconfig/types.go

```diff
@@ -160,58 +161,115 @@ const (
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
 
+// SchedulerPolicyConfigMapKey defines the key of the element in the
+// scheduler's policy ConfigMap that contains scheduler's policy config.
+const SchedulerPolicyConfigMapKey string = "policy.cfg"
+
+// SchedulerPolicySourceType define ways to obtain a scheduler Policy at
+// runtime.
+type SchedulerPolicySourceType string
+
+const (
+	// SchedulePolicySourceTypeFile means get a Policy from
+	// a SchedulerPolicyFileSource.
+	SchedulePolicySourceTypeFile SchedulerPolicySourceType = "file"
+	// SchedulerPolicySourceTypeConfigMap means get a Policy from a
+	// SchedulerPolicyConfigMapSource.
+	SchedulerPolicySourceTypeConfigMap SchedulerPolicySourceType = "configmap"
+)
+
+// SchedulerPolicySource configures a means to obtain a scheduler Policy. Given
+// a SchedulerPolicySourceType, the related source should be used. Sources are
+// mutually exclusive.
+type SchedulerPolicySource struct {
+	// Type is the policy source type.
+	Type SchedulerPolicySourceType
```

Comment:
unnecessary if you have a rule about "only one is non-nil", right?

Reply:
Yeah. Is omitting a "type" in a struct like this with mutually exclusive fields the preferred style?

Reply:
Removed Type.